test/new-e2e/tests/cws/fargate/fargate_test.go

```diff
@@ -0,0 +1,471 @@
+	DDCWSInstrumentationNamespace              = "cwsinstrumentation"
+	DDCWSInstrumentationFullImagePathParamName = "fullImagePath"
+)
+
+type ddCWSInstrumentationConfig struct {
+	CWSInstrumentationConfig *sdkconfig.Config
+	*configCommon.CommonEnvironment
+}
+
+func newCWSInstrumentationConfig(e *configCommon.CommonEnvironment) ddCWSInstrumentationConfig {
+	return ddCWSInstrumentationConfig{
+		sdkconfig.New(e.Ctx, DDCWSInstrumentationNamespace),
+		e,
+	}
+}
+
+func (c *ddCWSInstrumentationConfig) getFullImagePath() string {
+	return c.GetStringWithDefault(c.CWSInstrumentationConfig, DDCWSInstrumentationFullImagePathParamName, "docker.io/datadog/cws-instrumentation-dev:safchain-custom-cws-inst")
+}
+
+type ECSFargateSuite struct {
+	suite.Suite
+	ctx       context.Context
+	stackName string
+	testID    string
+
+	apiClient      *e2elib.APIClient
+	ddHostname     string
+	ecsClusterArn  string
+	ecsClusterName string
+	fgTaskDefArn   string
+}
+
+func TestECSFargate(t *testing.T) {
+	suite.Run(t, &ECSFargateSuite{
+		ctx:       context.Background(),
+		stackName: "cws-tests-ecs-fg",
+		testID:    e2elib.RandomString(4),
+	})
+}
+
+func (s *ECSFargateSuite) SetupSuite() {
+	s.apiClient = e2elib.NewAPIClient()
+
+	ruleDefs := []*e2elib.TestRuleDefinition{
+		{
+			ID:         execRuleID,
+			Expression: fmt.Sprintf(`exec.file.path == \"%s\"`, execFilePath),
+		},
+		{
+			ID:         openRuleID,
+			Expression: fmt.Sprintf(`open.file.path == \"%s\"`, openFilePath),
+		},
+	}
+	selftestsPolicy, err := e2elib.GetPolicyContent(ruleDefs)
+	s.Require().NoError(err)
+
+	_, result, err := infra.GetStackManager().GetStack(s.ctx, s.stackName, nil, func(ctx *pulumi.Context) error {
+		ddHostname := fmt.Sprintf("%s-%s", ddHostnamePrefix, s.testID)
+		awsEnv, err := awsResources.NewEnvironment(ctx)
+		if err != nil {
+			return err
+		}
+		cwsCfg := newCWSInstrumentationConfig(awsEnv.CommonEnvironment)
+
+		// Create cluster
+		ecsCluster, err := ecsResources.CreateEcsCluster(awsEnv, "cws-cluster")
+		if err != nil {
+			return err
+		}
+
+		// Export cluster’s properties
+		ctx.Export(ecsClusterNameKey, ecsCluster.Name)
+		ctx.Export(ecsClusterArnKey, ecsCluster.Arn)
+
+		// Associate Fargate capacity provider to the cluster
+		_, err = ecsResources.NewClusterCapacityProvider(awsEnv, "cws-cluster-capacity-provider", ecsCluster.Name, pulumi.StringArray{pulumi.String("FARGATE")})
+		if err != nil {
+			return err
+		}
+
+		// Setup agent API key
+		apiKeyParam, err := ssm.NewParameter(ctx, awsEnv.Namer.ResourceName("agent-apikey"), &ssm.ParameterArgs{
+			Name:  awsEnv.CommonNamer.DisplayName(1011, pulumi.String("agent-apikey")),
+			Type:  ssm.ParameterTypeSecureString,
+			Value: awsEnv.AgentAPIKey(),
+		}, awsEnv.WithProviders(configCommon.ProviderAWS, configCommon.ProviderAWSX))
+		if err != nil {
+			return err
+		}
+
+		// Create task definition
+		taskDef, err := ecsx.NewFargateTaskDefinition(ctx, "cws-task", &ecsx.FargateTaskDefinitionArgs{
```

Comment:
I changed to task definition to make use of the existing firelens container definition. It's a bit more complicated for the agent because we need to add environment variables, a volume, mountpoints and override the container entrypoint specifically for this test so doing this in the test-infra-definitions repository wouldn't make really sense.
Plus this keeps the task definition all in one place which makes it easier to read.

Reply:
I think you can use `GetFirelensLogConfiguration` at least. When we introduce `ECS` component we'll try to offer an API to handle these cases.